cmd/list.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2015 ThoughtWorks, Inc.
+
+// This file is part of Gauge.
+
+// Gauge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Gauge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Gauge.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"fmt"
+	supersort "sort"
+
+	"github.com/getgauge/gauge/gauge"
+	"github.com/getgauge/gauge/logger"
+	"github.com/getgauge/gauge/validation"
+	"github.com/spf13/cobra"
+)
+
+var (
+	listCmd = &cobra.Command{
+		Use:     "list [flags] [args]",
+		Short:   "List specifications, scenarios or tags for a gauge project",
+		Long:    `List specifications, scenarios or tags for a gauge project`,
+		Example: `  gauge list --tags specs`,
+		Run: func(cmd *cobra.Command, args []string) {
+			res := validation.ValidateSpecs(getSpecsDir(args), false)
+			if len(res.Errs) > 0 {
+				// how to print nicely errors ?
+				return
+			}
+			if specsFlag {
+				logger.Info(true, "[Specifications]")
+				listSpecifications(res.SpecCollection)
+			}
+			if scenariosFlag {
+				logger.Info(true, "[Scenarios]")
+				listScenarios(res.SpecCollection)
+			}
+			if tagsFlag {
+				logger.Info(true, "[Tags]")
+				listTags(res.SpecCollection)
+			}
+			if !specsFlag && !scenariosFlag && !tagsFlag {
+				exit(fmt.Errorf("Missing flag, nothing to list"), cmd.UsageString())
+			}
+		},
+		DisableAutoGenTag: true,
+	}
+	tagsFlag      bool
+	specsFlag     bool
+	scenariosFlag bool
+)
+
+func init() {
+	GaugeCmd.AddCommand(listCmd)
+	listCmd.Flags().BoolVarP(&tagsFlag, "tags", "t", false, "List the tags in projects")
```

Comment:
We had a discussion here and we would like to remove the shorthand flags for these commands.
This is mainly because the `--scenarios` flag should have a shorthand of `-s` too. But that would collide with `--specs` flag's shorthand.

Reply:
fine with me - changed :)